### PR TITLE
Removed references to grsim and cleaned up some bazel build rules

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Removed references to grsim and cleaned up some bazel build rules

### Testing Done
nothing functional changed

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/WORKSPACE [hunk 1]
```diff
@@ -116,7 +118,7 @@ http_archive(
     ],
 )
 
-# We use our own fork of Box2D as we have made a few small modifications
+# We use our ownfork of Box2D as we have made a few small modifications
```
Comment (akhilveeraghanta) [inline src/WORKSPACE]: nit: missing space

File: src/software/full_system_main.cpp [hunk 2]
```diff
@@ -121,6 +120,11 @@ int main(int argc, char **argv)
                 ->setValue(args.network_interface_name);
         }
 
+        if (args.backend_name.empty())
+        {
+            LOG(FATAL) << "The option '--backend' is required but missing";
+        }
+
```
Comment (akhilveeraghanta) [inline src/software/full_system_main.cpp:127]: Curious why this change is required?

Comment (jonathanlew) [reply src/software/full_system_main.cpp:127]: not required, but it could be a better error message than `"" is not a recognized backend`, now that `--backend` is not a required argument

**Review by garethellis0 — DISMISSED** [review]

File: src/WORKSPACE (hunk 1 above)
Comment (jonathanlew) [reply src/WORKSPACE]: fixed the typo

**Review by akhilveeraghanta — APPROVED** [review]
✔️

**Review by MathewMacDougall — APPROVED** [review]
Nice :+1:

File: src/WORKSPACE [hunk 3]
```diff
@@ -75,6 +75,7 @@ new_git_repository(
     build_file = "@//external:munkres_cpp.BUILD",
     commit = "61086fcf5b3a8ad4bda578cdc2d1dca57b548786",
     remote = "https://github.com/saebyn/munkres-cpp.git",
+    shallow_since = "1479502661 -0800",
```
Comment (jordanzliu) [inline src/WORKSPACE:78]: these don't seem like changes related to removing grSim

**Review by jordanzliu — CHANGES_REQUESTED** [review]
one minor comment, otherwise LGTM

**Review by jordanzliu — APPROVED** [review]
LGTM, `shallow_since` wasn't anything to worry about
